### PR TITLE
datacite minting fixes

### PR DESCRIPTION
**Pull request by blaiszik**
- Added ResourceTypeGeneral which is required for DOI minting
- Added `"event": "publish"` to mint the DOI directly instead cofminting a draft DOI

### CODE REVIEW

File: automate/minimus_mdf_flow.py [hunk 1]
```diff
@@ -335,6 +335,10 @@ def mint_doi_steps():
                         "publicationYear.$": "$.dataset_mdata.dc.publicationYear",
                         "url.$": "$.mdf_portal_link",
                     },
+                    "event": "publish",
```
Comment (BenGalewsky) [inline automate/minimus_mdf_flow.py]: I think these need to go up in the `attributes` section, and not in the `Doi` parent?

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
I think my understanding of the DataCite AP is correct, but please double check

**Comment by blaiszik** [comment]
Ok, I have made the requested changed @BenGalewsky. Thanks for catching that.

**Review by BenGalewsky — APPROVED** [review]
Great - let's try this out

**Pull request merged**